File: frontend/lib/models/comment.dart
class Comment {
  const Comment({
    required this.id,
    required this.postId,
    required this.authorUserId,
    required this.authorNickname,
    required this.content,
    this.replyToCommentId,
    this.replyToNickname,
    this.createdAt,
  });

  final String id;
  final String postId;
  final int authorUserId;
  final String authorNickname;
  final String content;
  final String? replyToCommentId;
  final String? replyToNickname;
  final DateTime? createdAt;

  factory Comment.fromJson(Map<String, dynamic> json) => Comment(
    id: json['id'] as String,
    postId: json['postId'] as String,
    authorUserId: json['authorUserId'] as int,
    authorNickname: json['authorNickname'] as String? ?? '社区用户',
    content: json['content'] as String,
    replyToCommentId: json['replyToCommentId'] as String?,
    replyToNickname: json['replyToNickname'] as String?,
    createdAt: DateTime.tryParse(json['createdAt'] as String? ?? ''),
  );
}

File: backend/src/main/java/com/campusmeow/acceptance/forum/document/Comment.java
package com.campusmeow.acceptance.forum.document;

import java.time.Instant;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("comments")
public class Comment {
    @Id
    private String id;
    private String postId;
    private Long authorUserId;
    private String content;
    private String replyToCommentId;
    private Instant createdAt;

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getPostId() { return postId; }
    public void setPostId(String postId) { this.postId = postId; }
    public Long getAuthorUserId() { return authorUserId; }
    public void setAuthorUserId(Long authorUserId) { this.authorUserId = authorUserId; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getReplyToCommentId() { return replyToCommentId; }
    public void setReplyToCommentId(String replyToCommentId) { this.replyToCommentId = replyToCommentId; }
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
}
